src/index.rs

```diff
@@ -290,15 +290,15 @@ pub fn add_document_to_index<T: Eq + Hash + Copy, D>(
     tokenizer: Tokenizer,
     filter: Filter,
     key: T,
-    doc: D,
+    doc: &D,
 ) {
     let docs = &mut index.docs;
     let fields = &mut index.fields;
     let mut field_length = vec![0; fields.len()];
     let mut term_counts: HashMap<String, Vec<usize>> = HashMap::new();
```

Comment:
While we are at it, we can make it even more performant: 

```suggestion
    let mut term_counts: HashMap<&str, Vec<usize>> = HashMap::new()
```